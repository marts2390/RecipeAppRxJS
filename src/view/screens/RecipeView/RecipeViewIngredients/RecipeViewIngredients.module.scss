@import "src/view/styles/variables.scss";

.ingredientsContainer {
  .ingredientsList {
    padding: 1.5em;
    border-radius: 10px;
    background-color: $white;

    .ingredientsItem {
      p {
        font-size: $font-xs;
        font-weight: $font-medium;
        margin: 0;
      }
    }

    h4 {
      margin-top: 0;
      margin-bottom: 0.75em;
      font-size: $font-lg;
      font-weight: $font-regular;
    }
  }

  .instructions {
    padding: 1.5em;
    border-radius: 10px;
    background-color: $white;

    p {
      font-weight: $font-medium;
    }

    h4 {
      margin-bottom: 0.75em;
      margin-top: 0;
      font-size: $font-lg;
      font-weight: $font-regular;
    }
  }
}
