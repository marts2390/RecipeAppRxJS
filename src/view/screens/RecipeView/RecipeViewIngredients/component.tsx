import React, { FC } from 'react'

import { Grid, Fade } from '@material-ui/core'

import styles from './RecipeViewIngredients.module.scss'

interface IRecipeViewIngredientsComponentProps {
  ingredientsList: string[]
  recipeMethod: string | null;
}

const RecipeViewIngredientsComponent: FC<IRecipeViewIngredientsComponentProps> = ({
  ingredientsList, recipeMethod,
}) => (
  <Fade
    in
    timeout={ 500 }
  >
    <div className={ styles.ingredientsContainer }>
      <Grid
        container
        spacing={ 2 }
      >
        <Grid
          item
          xs={ 12 }
          md={ 3 }
        >
          <div className={ styles.ingredientsList }>
            <h4>Ingredients</h4>
            <div className={ styles.ingredientsItem }>
              <Grid
                container
                spacing={ 2 }
              >
                {ingredientsList.map((item) => (
                  <Grid
                    item
                    key={ item }
                    xs={ 6 }
                    sm={ 4 }
                    md={ 12 }
                  >
                    <p>{item}</p>
                  </Grid>
                ))}
              </Grid>
            </div>
          </div>
        </Grid>
        <Grid
          item
          xs={ 12 }
          md={ 9 }
        >
          <div className={ styles.instructions }>
            <h4>Method</h4>
            <p>{recipeMethod}</p>
          </div>
        </Grid>
      </Grid>
    </div>
  </Fade>
)

export default RecipeViewIngredientsComponent
