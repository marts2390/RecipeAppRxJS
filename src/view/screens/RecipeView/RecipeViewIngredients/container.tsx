import React, { FC } from 'react'

import { useObservable } from 'rxjs-hooks'
import UC from 'actions'

import RecipeViewIngredientsComponent from './component'

const RecipeViewIngredientsContainer:FC = () => {
  const { recipeData, ingredientsList, isLoaded } = useObservable(UC.RecipeScreenService.getSubject)
|| UC.RecipeScreenService.getCurrentState()

  return (
    <RecipeViewIngredientsComponent
      ingredientsList={ ingredientsList }
      recipeMethod={ recipeData && recipeData.strInstructions }
      isLoaded={ isLoaded }
    />
  )
}

export default RecipeViewIngredientsContainer
