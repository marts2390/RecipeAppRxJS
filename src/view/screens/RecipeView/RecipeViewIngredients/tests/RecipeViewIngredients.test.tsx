import React from 'react'
import { shallow } from 'enzyme'
import RecipeViewIngredients from '../component'

let wrapper
let props

beforeEach(() => {
  props = {
    ingredientsList: ['ingredient1', 'ingredient2', 'ingredient3'],
    recipeMethod: 'This is the recipe method',
    isLoaded: false,
  }

  wrapper = shallow(<RecipeViewIngredients { ...props } />)
})

test('Ingredients list is rendered', () => {
  expect(wrapper).toMatchSnapshot()
})

test('isLoading is false', () => {
  const elem = wrapper.find({ 'data-testid': 'ingredientWrapper' })
  expect(elem.prop('in')).toBe(false)
})

test('isLoading is true', () => {
  wrapper.setProps({ isLoaded: true })
  const elem = wrapper.find({ 'data-testid': 'ingredientWrapper' })
  expect(elem.prop('in')).toBe(true)
})
