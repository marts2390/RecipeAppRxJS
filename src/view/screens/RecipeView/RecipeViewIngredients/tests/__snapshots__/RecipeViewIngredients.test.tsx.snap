// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients list is rendered 1`] = `
<ForwardRef(Fade)
  data-testid="ingredientWrapper"
  in={false}
  timeout={500}
>
  <div
    className="ingredientsContainer"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={3}
        xs={12}
      >
        <div
          className="ingredientsList"
        >
          <h4>
            Ingredients
          </h4>
          <div
            className="ingredientsItem"
          >
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                key="ingredient1"
                md={12}
                sm={4}
                xs={6}
              >
                <p>
                  ingredient1
                </p>
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                key="ingredient2"
                md={12}
                sm={4}
                xs={6}
              >
                <p>
                  ingredient2
                </p>
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                key="ingredient3"
                md={12}
                sm={4}
                xs={6}
              >
                <p>
                  ingredient3
                </p>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={9}
        xs={12}
      >
        <div
          className="instructions"
        >
          <h4>
            Method
          </h4>
          <p>
            This is the recipe method
          </p>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</ForwardRef(Fade)>
`;
